core/model/integration.py

```diff
@@ -58,45 +78,44 @@ class IntegrationConfiguration(Base):
         "IntegrationLibraryConfiguration",
         back_populates="parent",
         uselist=True,
-        cascade="all, delete",
+        cascade="all, delete-orphan",
         passive_deletes=True,
     )
 
-    collection: Mapped[Collection] = relationship("Collection", uselist=False)
+    collection: Mapped[Collection] = relationship(
+        "Collection", back_populates="integration_configuration", uselist=False
+    )
 
-    @overload
-    def for_library(
-        self, library_id: int, create: Literal[True]
-    ) -> IntegrationLibraryConfiguration:
-        ...
+    # https://docs.sqlalchemy.org/en/14/orm/extensions/associationproxy.html#simplifying-association-objects
+    libraries: Mapped[List[Library]] = association_proxy(
+        "library_configurations",
+        "library",
+        creator=lambda library: IntegrationLibraryConfiguration(library=library),
+    )
```

Comment:
This makes it easy to access the libraries for this integration, if you don't need to access the intermediate settings object.